src/CalDAV/Backend/Calendar.php

```diff
@@ -216,7 +216,10 @@ public function getCalendarObject($calendarId, $objectPath)
             return null;
         }
 
-        $vcalendar = $item->getAsVCalendar();
+        $vcalendar = null;
+        if (method_exists($item, 'getAsVCalendar')) {
```

Comment:
Can check for interface `CalDAVCompatibleItemInterface` instead.

Reply:
`getCalendarItemForPath` is already supposed to return `CalDAVCompatibleItemInterface|null`.

Changing `if (null === $item) {` to `if (! ($item instanceof CalDAVCompatibleItemInterface)) {` whould be a better fix indeed.

It is weird that PHPStan does not handle this case properly.

Reply:
Some changes may need made to `getCalendarItemForUid`. My IDE says the return type is inconsistent with that method. It thinks the return type is CommonDBTM or null instead of the interface or null. Maybe that is affecting the PHPStan check for `getCalendarItemForPath` which calls this.

Reply:
Maybe something like:
```diff
diff --git a/src/CalDAV/Traits/CalDAVUriUtilTrait.php b/src/CalDAV/Traits/CalDAVUriUtilTrait.php
index db0f5283b2..33932f731f 100644
--- a/src/CalDAV/Traits/CalDAVUriUtilTrait.php
+++ b/src/CalDAV/Traits/CalDAVUriUtilTrait.php
@@ -245,11 +245,10 @@ trait CalDAVUriUtilTrait
         }
 
         $item_specs = $items_iterator->current();
-        if (!is_a($item_specs['itemtype'], CalDAVCompatibleItemInterface::class, true)) {
-            return null;
-        }
 
-        if (!$item = getItemForItemtype($item_specs['itemtype'])) {
+        $item = getItemForItemtype($item_specs['itemtype']);
+
+        if (!($item instanceof CalDAVCompatibleItemInterface)) {
             return null;
         }
```